src/Battlescape/SkillMenuState.cpp

```diff
@@ -209,8 +215,7 @@ void SkillMenuState::btnActionMenuItemClick(Action *action)
 
 		if (!continueAction || _action->type == BA_NONE)
 		{
-			_action->type = BA_NONE;
-			_action->targeting = false;
+			_action->clear();
```

Comment:
Could you copy this change to other places too? If some code reset `type` and `targeting` it should be replaced by `clear()`